### PR TITLE
GIPF-33: Integrate ESLint into CI for Pull Requests

### PR DESCRIPTION
**Pull request by adrianHards**
This commit introduces a new workflow to our CI process. Each time a PR is raised, the workflow ensures that our TypeScript code adheres to the established linting standards by running ESLint checks. This integration enhances code quality and consistency throughout the project.

### CODE REVIEW

**Comment by adrianHards** [comment]
[Here's](https://github.com/Loo-Ashworth/acebook-poke/blob/main/.github/workflows/main.yml#L1C1-L8C13) an example of a previous workflow I helped create @sulphite

File: .github/workflows/eslint.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: ESLint
+
+on:
+  pull_request:
+    paths:
+      - '**.ts'
+      - '.eslintrc.js'
+      - 'pnpm-lock.yaml'
```
Comment (sulphite) [inline .github/workflows/eslint.yml]: I've set the linter to only check files in the src directory, so I'm thinking this should be the same?

Comment (adrianHards) [reply .github/workflows/eslint.yml]: modified to `- 'src/**/*.ts'`

**Review by sulphite — APPROVED** [review]
looks good hope it works

**Pull request merged**